python_modules/dagster/dagster_tests/general_tests/py3_tests/test_inference.py

```diff
@@ -575,3 +575,30 @@ def add_one(a: MyClass) -> MyClass:
         return subtract(a, emit_one())
 
     assert add_one.input_mappings
+
+
+def test_none_return_annotation_to_nothing():
+    @op
+    def my_op() -> None:
+        ...
+
+    assert my_op.output_defs[0].dagster_type.is_nothing
+
+
+def test_none_param_annotation_to_nothing():
+    @op
+    def my_op(_in1: None, _in2: None, _in3: str):
+        ...
+
+    assert my_op.input_defs[0].dagster_type.is_nothing
+    assert my_op.input_defs[1].dagster_type.is_nothing
+    assert not my_op.input_defs[2].dagster_type.is_nothing
+
+
+def test_empty_annotations_to_not_nothing():
+    @op
+    def my_op(_in1):
+        ...
+
+    assert not my_op.output_defs[0].dagster_type.is_nothing
+    assert not my_op.input_defs[0].dagster_type.is_nothing
```

Comment:
if someone has something like 

```
@op
def produce() -> None:
    pass


@op
def consume(x):
    return x


@graph
def breaking_change():
    consume(produce())
```

then this will be a breaking change for them